{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Names</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Significance</th>\n",
       "      <th>variant_type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">benign</th>\n",
       "      <th>duplication</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_000268.3(NF2):c.*2032dupT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>insertion</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_000268.3(NF2):c.*2032dupT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snp</th>\n",
       "      <td>18</td>\n",
       "      <td>NM_000268.3(NF2):c.-246C&gt;G, NM_000268.3(NF2):c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>benign/likely benign</th>\n",
       "      <th>snp</th>\n",
       "      <td>3</td>\n",
       "      <td>NM_000268.3(NF2):c.12C&gt;T (p.Ala4=), NM_000268....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>conflicting interpretations of pathogenicity</th>\n",
       "      <th>snp</th>\n",
       "      <td>3</td>\n",
       "      <td>NM_000268.3(NF2):c.240+15C&gt;T, NM_000268.3(NF2)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">likely benign</th>\n",
       "      <th>deletion</th>\n",
       "      <td>2</td>\n",
       "      <td>NM_000268.3(NF2):c.*704delA, NM_000268.3(NF2):...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snp</th>\n",
       "      <td>53</td>\n",
       "      <td>NM_181832.2(NF2):c.-18G&gt;A, NM_000268.3(NF2):c....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>likely pathogenic</th>\n",
       "      <th>deletion</th>\n",
       "      <td>2</td>\n",
       "      <td>NC_000022.11:g.(?_29668333)_(29671948_?)del, N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>not provided</th>\n",
       "      <th>snp</th>\n",
       "      <td>2</td>\n",
       "      <td>NM_000268.3(NF2):c.1007G&gt;A (p.Arg336Gln), NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">pathogenic</th>\n",
       "      <th>deletion</th>\n",
       "      <td>10</td>\n",
       "      <td>NM_000268.3(NF2):c.-443-?_*3798+?del, NM_00026...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>duplication</th>\n",
       "      <td>9</td>\n",
       "      <td>NC_000022.10:g.(?_16054691)_(51237463_?)dup, N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>insertion</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_000268.3(NF2):c.125_126insG (p.Gly43Argfs)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snp</th>\n",
       "      <td>20</td>\n",
       "      <td>NM_000268.3(NF2):c.169C&gt;T (p.Arg57Ter), NM_000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">uncertain significance</th>\n",
       "      <th>deletion</th>\n",
       "      <td>5</td>\n",
       "      <td>NM_000268.3(NF2):c.1739_1747delTCACCTTGC (p.Le...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>duplication</th>\n",
       "      <td>11</td>\n",
       "      <td>NM_000268.3(NF2):c.*768_*769dupAG, NM_000268.3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>insertion</th>\n",
       "      <td>7</td>\n",
       "      <td>NM_000268.3(NF2):c.*768_*769dupAG, NM_000268.3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snp</th>\n",
       "      <td>92</td>\n",
       "      <td>NM_000268.3(NF2):c.-402A&gt;G, NM_000268.3(NF2):c...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                          Count  \\\n",
       "Significance                                 variant_type         \n",
       "benign                                       duplication      1   \n",
       "                                             insertion        1   \n",
       "                                             snp             18   \n",
       "benign/likely benign                         snp              3   \n",
       "conflicting interpretations of pathogenicity snp              3   \n",
       "likely benign                                deletion         2   \n",
       "                                             snp             53   \n",
       "likely pathogenic                            deletion         2   \n",
       "not provided                                 snp              2   \n",
       "pathogenic                                   deletion        10   \n",
       "                                             duplication      9   \n",
       "                                             insertion        1   \n",
       "                                             snp             20   \n",
       "uncertain significance                       deletion         5   \n",
       "                                             duplication     11   \n",
       "                                             insertion        7   \n",
       "                                             snp             92   \n",
       "\n",
       "                                                                                                       Names  \n",
       "Significance                                 variant_type                                                     \n",
       "benign                                       duplication                        NM_000268.3(NF2):c.*2032dupT  \n",
       "                                             insertion                          NM_000268.3(NF2):c.*2032dupT  \n",
       "                                             snp           NM_000268.3(NF2):c.-246C>G, NM_000268.3(NF2):c...  \n",
       "benign/likely benign                         snp           NM_000268.3(NF2):c.12C>T (p.Ala4=), NM_000268....  \n",
       "conflicting interpretations of pathogenicity snp           NM_000268.3(NF2):c.240+15C>T, NM_000268.3(NF2)...  \n",
       "likely benign                                deletion      NM_000268.3(NF2):c.*704delA, NM_000268.3(NF2):...  \n",
       "                                             snp           NM_181832.2(NF2):c.-18G>A, NM_000268.3(NF2):c....  \n",
       "likely pathogenic                            deletion      NC_000022.11:g.(?_29668333)_(29671948_?)del, N...  \n",
       "not provided                                 snp           NM_000268.3(NF2):c.1007G>A (p.Arg336Gln), NM_0...  \n",
       "pathogenic                                   deletion      NM_000268.3(NF2):c.-443-?_*3798+?del, NM_00026...  \n",
       "                                             duplication   NC_000022.10:g.(?_16054691)_(51237463_?)dup, N...  \n",
       "                                             insertion         NM_000268.3(NF2):c.125_126insG (p.Gly43Argfs)  \n",
       "                                             snp           NM_000268.3(NF2):c.169C>T (p.Arg57Ter), NM_000...  \n",
       "uncertain significance                       deletion      NM_000268.3(NF2):c.1739_1747delTCACCTTGC (p.Le...  \n",
       "                                             duplication   NM_000268.3(NF2):c.*768_*769dupAG, NM_000268.3...  \n",
       "                                             insertion     NM_000268.3(NF2):c.*768_*769dupAG, NM_000268.3...  \n",
       "                                             snp           NM_000268.3(NF2):c.-402A>G, NM_000268.3(NF2):c...  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('/home/data/clinvar_result_combined_last.txt', delimiter=\"\\t\")\n",
    "\n",
    "select_cols = df[['Name',\n",
    "                  'Review status',\n",
    "                  'AlleleID(s)',\n",
    "                  'VariationID',\n",
    "                  'Gene(s)',\n",
    "                  'GRCh38Chromosome',\n",
    "                  'GRCh38Location',\n",
    "                  'AlleleID(s)',\n",
    "                  'variant_type'\n",
    "                 ]]\n",
    "\n",
    "# strip \"(Last reviewed.*\" from each 'Clinical significance' record\n",
    "significance = df['Clinical significance (Last reviewed)']\n",
    "significance_stripped = significance.str.lower().str.replace(r'\\(last reviewed.*', '')\n",
    "select_cols = select_cols.assign(Significance = significance_stripped)\n",
    "\n",
    "# group by\n",
    "agg_funcs = {\n",
    "             'Name': {'Names': [lambda i: \", \".join(i)]},\n",
    "             #'variant_type': [lambda i: \", \".join(set(i))],\n",
    "             'VariationID': {'Count': [lambda i: str(len([str(j) for j in i]))]}\n",
    "            }\n",
    "\n",
    "x = select_cols.groupby(['Significance', 'variant_type']).agg(agg_funcs)\n",
    "x.columns = x.columns.droplevel(1)\n",
    "x\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
